### ISSUE TITLE
Add more loss functions

### ISSUE THREAD

**Issue by aromanro** [aromanro/MachineLearning#3]
For now only L2, L1 and binary cross-entropy losses are implemented.
Should be enough for simple examples/cases... but more could be added (as in Huber and so on).

This is postponed until the project reaches a functional state (up to the intended multilayer perceptron).
